handlers.go

```diff
@@ -96,7 +119,25 @@ func RPCInvokeHandlerWithOptions(ch grpc.ClientConnInterface, descs []*desc.Meth
 					http.Error(w, "Failed to create descriptor source: "+err.Error(), http.StatusInternalServerError)
 					return
 				}
-				results, err := invokeRPC(r.Context(), method, ch, descSource, r.Header, r.Body, &options)
+				req := &RPCRequest{
+					MethodName: method,
+					Conn:       ch,
+					DescSource: descSource,
+					Headers:    r.Header,
+					Body:       r.Body,
+					Options:    &options,
+				}
+				call := func(ctx context.Context, req *RPCRequest) (*RPCResult, error) {
+					return invokeRPC(ctx, method, ch, descSource, r.Header, r.Body, &options)
+				}
+				for i := len(options.Middlewares) - 1; i > 0; i-- {
+					mw := options.Middlewares[i]
+					c2 := call
+					call = func(ctx context.Context, req *RPCRequest) (*RPCResult, error) {
+						return mw(ctx, req, c2)
+					}
+				}
+				results, err := call(r.Context(), req)
```

Comment:
There's something slightly off for me about the local variable names here, but I can't put my finger on it.  Lemme play with this a minute.